packages/react/src/context/slash-id-context.tsx

```diff
@@ -321,12 +350,22 @@ export const SlashIDProvider = ({
     };
 
     const tryImmediateLogin = async () => {
-      if (token) {
-        storeUser(new User(token, sidRef.current!));
+      if (token && (await validateToken(token))) {
+        const user = new User(token, sid);
+
+        if (user.anonymous && anonymousUsersEnabled) {
+          const anonUser = new AnonymousUser(token, sid);
+          storeUser(anonUser);
+        } else {
+          storeUser(user);
+        }
       } else {
         const isDone = await loginDirectIdIfPresent();
 
-        if (!isDone) {
+        if (!isDone && anonymousUsersEnabled) {
+          const user = await sid.createAnonymousUser();
+          storeUser(user);
+        } else if (!isDone) {
```

Comment:
@jake-slashid if I'm reading this right, enabling anonymous users will prevent the SDK from reading tokens from storage.

We should only create an anonymous user if we have already looked into the storage and found it empty.

This function was a bit messy before adding anonymous users, I was wondering if we could refactor it in a way that makes it clear what the priority is? I've had success in previous projects following the chain of responsibility pattern:

```
const chain = [
 useTokenFromProps,
 useDirectID,
 useStoredToken,
 useAnonymousUser
]

// execute chain in sequence
// stop execution once the first function returns a truthy value
// each function can either return a Promise that resolves to a User or resolve with `false` or similar
```
This way the priority and the sequence of steps is made clear. What do you think?

Reply:
I agree that the reasonability of this block is very poor, but its intended that the anon user creation is last.

Reply:
I’ll refactor this function

Reply:
done

There is a new utility: `sequence()`